// fake-products-sj/index.js
var products = [
  {
    id:0,
    name: "Iphone X",
    amount: 3
  },
  {
    id:1,
    name: "Iphone 8",
    amount: 7
  },
  {
    id:2,
    name: "Samsung Note",
    amount: 1
  },
  {
    id:3,
    name: "Ipad pro",
    amount: 4
  }
]
var counter = products.length;
var historyProducts = [];


exports.getProducts = function() {
  return products;
}

exports.getHistorySell = function() {
  return historyProducts;
}

exports.addProduct = function(product) {
    product.id = counter++;
    products.push(product);
}

exports.deleteProduct = function(id) {
    for(var i=0; i<products.length; i++){
      if (products[i].id === id) {
          return products.splice(i, 1);
      }
    }
}

exports.editProduct = function(id, newName, newAmount) {
    for(var i=0; i<products.length; i++){
      if (products[i].id === id) {
          products[i].amount = newAmount;
          products[i].name = newName;
      }
    }
}

exports.buyProduct = function(id, token) {
    for(var i=0; i<products.length; i++){
      if (products[i].id === id) {
          if (products[i].amount > 0) {
              products[i].amount--;
              return historyProducts.push({
                idProduct:products[i].id,
                tokenUser: token,
                timestamp: new Date()
              })

          }
      }
    }
}
